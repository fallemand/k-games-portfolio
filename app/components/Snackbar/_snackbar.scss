@import '../../styles/variables';
@import 'snackbar__small';

@media ( min-width: $breakpoint-desktop ) {
  @import 'snackbar__medium';
}

