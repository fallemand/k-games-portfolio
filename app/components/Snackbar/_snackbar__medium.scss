@import '../../styles/variables';

.snackbar {
  width: 400px;
  margin-left: -200px;
  padding: 16px;
  left: 50%;
  bottom: 80px;

  .snackbar__message {
    font-size: $font-size-16;
  }

  @-webkit-keyframes fadein {
    from {bottom: 0; opacity: 0;}
    to {bottom: 80px; opacity: 1;}
  }

  @keyframes fadein {
    from {bottom: 0; opacity: 0;}
    to {bottom: 80px; opacity: 1;}
  }

  @-webkit-keyframes fadeout {
    from {bottom: 80px; opacity: 1;}
    to {bottom: 0; opacity: 0;}
  }

  @keyframes fadeout {
    from {bottom: 80px; opacity: 1;}
    to {bottom: 0; opacity: 0;}
  }
}
