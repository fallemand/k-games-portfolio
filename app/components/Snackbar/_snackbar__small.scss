@import '../../styles/variables';

.snackbar {
  visibility: hidden;
  width: 100%;
  background-color: $background-color-secondary;
  text-align: center;
  border-radius: $border-radius;
  padding: 30px;
  position: fixed;
  z-index: 1;
  left: 0;
  bottom: 0;

  .snackbar__message {
    color: $font-color-secondary;
    font-size: $font-size-18;
  }

  &.snackbar--show {
    visibility: visible;
    -webkit-animation: fadein 0.5s, fadeout 0.5s 2.5s;
    animation: fadein 0.5s, fadeout 0.5s 2.5s;
  }

  @-webkit-keyframes fadein {
    from {bottom: -80px; opacity: 0;}
    to {bottom: 0; opacity: 1;}
  }

  @keyframes fadein {
    from {bottom: -80px; opacity: 0;}
    to {bottom: 0; opacity: 1;}
  }

  @-webkit-keyframes fadeout {
    from {bottom: 0; opacity: 1;}
    to {bottom: -80px; opacity: 0;}
  }

  @keyframes fadeout {
    from {bottom: 0; opacity: 1;}
    to {bottom: -80px; opacity: 0;}
  }
}
