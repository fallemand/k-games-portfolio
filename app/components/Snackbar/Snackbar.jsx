import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './_snackbar.scss';

const Snackbar = ({ className, show, message }) => (
  <div
    role="alert"
    className={classnames(className, 'snackbar', { 'snackbar--show': show })}
  >
    <span className="snackbar__message">
      {message}
    </span>
  </div>
);

Snackbar.propTypes = {
  className: PropTypes.string,
  show: PropTypes.bool,
  message: PropTypes.string,
};

Snackbar.defaultProps = {
  className: null,
  show: false,
  message: '',
};

export default Snackbar;
