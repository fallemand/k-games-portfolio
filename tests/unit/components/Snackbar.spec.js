import React from 'react';
import Snackbar from '../../../app/components/Snackbar';

describe('Button', () => {
  const props = {
    className: '__CLASSNAME__',
    message: '__MESSAGE__',
  };

  it('render with show:true', () => {
    const component = shallow(<Snackbar {...props} show />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('render with show:false', () => {
    const component = shallow(<Snackbar {...props} show={false} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
