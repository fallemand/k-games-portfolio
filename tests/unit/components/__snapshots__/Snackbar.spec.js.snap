// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with show:false 1`] = `
"<div role='alert' className='__CLASSNAME__ snackbar'>
  <span className='snackbar__message'>
    __MESSAGE__
  </span>
</div>"
`;

exports[`Button render with show:true 1`] = `
"<div role='alert' className='__CLASSNAME__ snackbar snackbar--show'>
  <span className='snackbar__message'>
    __MESSAGE__
  </span>
</div>"
`;
